internal/elasticsearch/cluster/component_template.go

```diff
@@ -0,0 +1,290 @@
+package cluster
```

Comment:
Could you, please, move this file into `internal/elasticsearch/index/` folder (`index` module), where the `template` resource is located, just to keep them all together?